src/Components/ADT3DSceneBuilder/Internal/Behaviors/Widgets/WidgetLibraryDialog/WidgetLibraryDialog.tsx

```diff
@@ -51,6 +60,107 @@ const WidgetLibraryDialog: React.FC<{
         }
     };
 
+    const isSpinnerVisible = useCallback(
+        (widgetType: WidgetType) =>
+            widgetType === WidgetType.DataHistory &&
+            scenePageContext?.state.adxConnectionInformation?.loadingState ===
+                ADXConnectionInformationLoadingState.LOADING,
+        [scenePageContext?.state.adxConnectionInformation?.loadingState]
+    );
+
+    const isWidgetNotAvailable = useCallback(
+        (widgetType: WidgetType) =>
+            widgetType === WidgetType.DataHistory &&
+            scenePageContext?.state.adxConnectionInformation?.loadingState ===
+                ADXConnectionInformationLoadingState.NOT_EXIST,
+        [scenePageContext?.state.adxConnectionInformation?.loadingState]
+    );
+
+    const handleOnRenderCell = useCallback(
+        (widget: IWidgetLibraryItem, index: number) => (
+            <DefaultButton
+                disabled={
+                    widget.data.type === WidgetType.DataHistory &&
+                    scenePageContext?.state.adxConnectionInformation
+                        ?.loadingState !==
+                        ADXConnectionInformationLoadingState.EXIST
+                }
+                key={index}
+                className={css(
+                    'cb-widget-dialog-list-item',
+                    index === selectedWidget
+                        ? 'cb-widget-dialog-list-item-selected'
+                        : ''
+                )}
+                onClick={() => {
+                    setSelectedWidget(index);
+                    setFilteredAvailableWidgets([...enabledWidgets]);
+                }}
+                data-testid={`widget-library-${widget.data.type}`}
+                styles={{
+                    flexContainer: { justifyContent: 'start' }
+                }}
+                selected={index === selectedWidget}
+            >
+                <Stack horizontal>
+                    <Stack
+                        className="cb-widget-dialog-icon-container"
+                        aria-hidden={true}
+                    >
+                        {isSpinnerVisible(widget.data.type as WidgetType) ? (
```

Comment:
it happens because in schema file there is no separate type defined for widget type, but it is list of strings. But on the components side we use a strict typing for it which is defined in 3DVConfig.ts (which we use for additional typings on top of schema generated typings). That is why I need to do this casting. I can update the schema to generate a separate type and remove the one we use in 3DVConfig.ts:
![image](https://user-images.githubusercontent.com/45217314/195214031-abf7e828-af76-43ff-a9ff-18cd7a9a69b4.png)

Reply:
in fact I don't know how to define a string const using an enum in json schema. I think that is why we used a const string for "type" of widgets and define an enum based on it in 3DVConfig.ts. So, I will leave this as is for now.